otumat/usage.py

```diff
@@ -0,0 +1,528 @@
+            f.write(f'{cmd} &>/dev/null &\n')
+    elif general_system() == 'Darwin':
+        # trigger startup using launchd by utiling launch agents
+        makedirs(Path(home_dir, 'Library', 'LaunchAgents'), exist_ok=True)
+        with open(Path(home_dir, 'Library', 'LaunchAgents',
+                  f'{package_name}_usage.startup.plist'), 'w') as f:
+            f.write(f"""
+            <?xml version="1.0" encoding="UTF-8"?>
+            <!DOCTYPE plist PUBLIC
+                "-//Apple//DTD PLIST 1.0//EN"
+                "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+            <plist version="1.0">
+            <dict>
+                <key>EnvironmentVariables</key>
+                <dict>
+                    <key>PATH</key>
+                    <string>{getenv('PATH')}</string>
+                </dict>
+                <key>Label</key>
+                <string>{package_name}_usage.startup</string>
+                <key>RunAtLoad</key>
+                <true/>
+                <key>ProgramArguments</key>
+                <array>
+                    {''.join([f'<string>{t}</string>' for t in ('bash', '-c', cmd)])}
+                </array>
+            </dict>
+            </plist>
+            """)
+    elif general_system() == 'Windows':
+        # trigger startup by utilizing user's startup directory and a VBScript to start a
+        # background program
+        with open(Path(home_dir, 'AppData', 'Roaming', 'Microsoft', 'Windows', 'Start Menu',
+                       'Programs', 'Startup', f'{package_name}_usage.vbs'), 'w') as f:
+            f.write(f"""
+            Dim WinScriptHost
+            Set WinScriptHost = CreateObject("WScript.Shell")
+            WinScriptHost.Run "{cmd}", 0
+            Set WinScriptHost = Nothing
+            """)
+
+
+def _deactivate_startup(package_name):
+    """
+    Utility that will unregister a daemon from running at startup.
+
+    :param package_name: Name of package to identify the daemon
+    :type package_name: str
+    """
+    home_dir = getenv('USERPROFILE', getenv('HOME'))
+    if general_system() == 'Linux':
+        # disable startup by removing daemon command from profile script
+        # Bourne shell compatible
+        startup_file = Path(home_dir, '.profile')
+        if startup_file.exists():
+            lines = open(startup_file, 'r').readlines()
+            with open(startup_file, 'w') as f:
+                [f.write(line) for line in lines if ('otumat' not in line and
+                                                     'upload' not in line and
+                                                     package_name not in line)]
+    elif general_system() == 'Darwin':
+        # disable startup by removing property list file spec for the launch agent
+        startup_file = Path(home_dir, 'Library', 'LaunchAgents',
+                            f'{package_name}_usage.startup.plist')
+        if startup_file.exists():
```

Comment:
don't check if exists first: unlink and catch error -- it's generally a better process for race conditions, etc.

Reply:
Ah yes, good point on race condition. I've updated it now to simply try to delete can catch the exception.